app/models/in_progress_form.rb

```diff
@@ -93,29 +93,44 @@ def expires_after
   end
 
   def log_cfi_metric(params)
-    if form_id == '21-526EZ' && metadata['cfiMetric'].blank?
-      fd = params[:form_data] || params[:form_Data]
-      fd = JSON.parse(fd) if fd.is_a?(String)
-      if fd.present? &&
-         (fd['rated_disabilities'].present? || fd['ratedDisabilities'].present?) &&
-         (fd.dig('view:claim_type', 'view:claiming_increase') || fd.dig('view:claimType', 'view:claimingIncrease'))
-
-        max_cfi_enabled = Flipper.enabled?(:disability_526_maximum_rating) ? 'on' : 'off'
-
-        rated_disabilities = fd['rated_disabilities'] || fd['ratedDisabilities']
-        rated_disabilities.each do |dis|
-          if dis['maximum_rating_percentage'].present? || dis['maximumRatingPercentage'].present?
-            dis_name = dis['name'].parameterize(separator: '_')
-            StatsD.increment("#{MAX_CFI_STATSD_KEY_PREFIX}_#{max_cfi_enabled}_rated_disabilities_#{dis_name}")
-            params[:metadata]['cfiMetric'] = true
-          end
-        end
-      end
+    return unless form_id == '21-526EZ'
+
+    # if the cfiMetric is present in the form already, then the disability education has been seen
+    if metadata['cfiMetric'].present?
+      params[:metadata]['cfiMetric'] = true
```

Comment:
This addition of cfiMetric to true is to make sure the form is updated with this value set again. Otherwise the form is updated with only the metadata from the request.